mongodbatlas/resource_mongodbatlas_federated_database_instance.go

```diff
@@ -529,151 +517,127 @@ func newStores(d *schema.ResourceData) []*matlas.DataFederationStore {
 	return stores
 }
 
-func newAdditionalStorageClasses(additionalStorageClassesFromConfig []interface{}) []*string {
+func newAdditionalStorageClasses(additionalStorageClassesFromConfig []interface{}) []string {
 	if len(additionalStorageClassesFromConfig) == 0 {
 		return nil
 	}
 
-	additionalStorageClasses := make([]*string, len(additionalStorageClassesFromConfig))
+	additionalStorageClasses := make([]string, len(additionalStorageClassesFromConfig))
 	for i, additionalStorageClassFromConfig := range additionalStorageClassesFromConfig {
-		additionalStorageClasses[i] = pointer(additionalStorageClassFromConfig.(string))
+		additionalStorageClasses[i] = additionalStorageClassFromConfig.(string)
 	}
 
 	return additionalStorageClasses
 }
 
-func newReadPreference(storeFromConfMap map[string]interface{}) *matlas.ReadPreference {
+func newReadPreference(storeFromConfMap map[string]interface{}) *admin.DataLakeAtlasStoreReadPreference {
 	readPreferenceFromConf, ok := storeFromConfMap["read_preference"].([]interface{})
 	if !ok || len(readPreferenceFromConf) == 0 {
 		return nil
 	}
 	readPreferenceFromConfMap := readPreferenceFromConf[0].(map[string]interface{})
-	return &matlas.ReadPreference{
-		Mode:                readPreferenceFromConfMap["mode"].(string),
-		MaxStalenessSeconds: int32(readPreferenceFromConfMap["max_staleness_seconds"].(int)),
-		TagSets:             newTagSets(readPreferenceFromConfMap),
```

Comment:
Removing TagSets implementation because of difference in data type. Addressed in PR -